doctr/models/_utils.py

```diff
@@ -21,29 +21,35 @@ def get_max_width_length_ratio(contour: np.ndarray) -> float:
     ----
         contour: the contour from cv2.findContour
 
-    Returns: the maximum shape ratio
+    Returns:
+    -------
+        the maximum shape ratio
     """
     _, (w, h), _ = cv2.minAreaRect(contour)
     return max(w / h, h / w)
 
 
-def estimate_orientation(img: np.ndarray, n_ct: int = 50, ratio_threshold_for_lines: float = 5) -> float:
+def estimate_orientation(img: np.ndarray, n_ct: int = 50, ratio_threshold_for_lines: float = 5) -> int:
     """Estimate the angle of the general document orientation based on the
      lines of the document and the assumption that they should be horizontal.
 
     Args:
     ----
-        img: the img to analyze
+        img: the img or bitmap to analyze (H, W, C)
         n_ct: the number of contours used for the orientation estimation
         ratio_threshold_for_lines: this is the ratio w/h used to discriminates lines
 
     Returns:
     -------
         the angle of the general document orientation
     """
-    gray_img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
-    gray_img = cv2.medianBlur(gray_img, 5)
-    thresh = cv2.threshold(gray_img, thresh=0, maxval=255, type=cv2.THRESH_BINARY_INV + cv2.THRESH_OTSU)[1]
+    assert len(img.shape) == 3 and img.shape[-1] in [1, 3], f"Image shape {img.shape} not supported"
+    if np.max(img) <= 1 and np.min(img) >= 0 or (np.max(img) <= 255 and np.min(img) >= 0 and img.shape[-1] == 1):
+        thresh = img.astype(np.uint8)
+    if np.max(img) <= 255 and np.min(img) >= 0 and img.shape[-1] == 3:
```

Comment:
I think there can be a performance hit due to the `np.max` and `np.min`.
Here, you want to do an automatic distinction between image and grayscale image ?

Reply:
Correct the function can handle now rgb images and bitmaps

Reply:
What would you suggest instead ? ^^

Reply:
For bitmap, is it bool or float in [0,1] ? Is it possible to make an assumption on the type directly ?
First optimization can be to compute `np.max` and `np.min` once.

Reply:
yeah sure that's a good one :sweat_smile: 

Mh normally the function is only called with binary images atm 

-> `seg_maps = [np.where(out_map > kwargs.get("bin_thresh", 0.3), 255, 0).astype(np.uint8) for out_map in out_maps]`

but personally i would like to keep it a bit more flexible because it works also with RGB images

Reply:
At the end it's a decision we have to make we can lower the checks to support only binary images which is the only case atm or we keep it more flexible to work with different image types wdyt ?

Reply:
but computing it onces was a really good catch :sweat_smile: :+1:

Reply:
Mmmh, I'm not sure it will have such impact on performance in the end. Let's keep it like that.